File: app/src/main/java/ch/bailu/aat/map/layer/gpx/DistanceCounter.java
package ch.bailu.aat.map.layer.gpx;

public final class DistanceCounter {
    public final float min, max;
    private float count=0;

    public DistanceCounter(float limit1, float limit2) {
        max = Math.max(limit1, limit2);
        min = Math.min(limit1, limit2);
    }

    public void add(float x) {
        count += x;
    }


    public void reset() {
        count = 0;
    }


    public boolean isTooSmall() {
        return count < min;
    }

    public boolean hasDistance() {
        return count >= min;
    }


    public boolean isTooLarge() {
        return count > max;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/gpx/DistanceCounter.kt
package ch.bailu.aat_lib.map.layer.gpx

class DistanceCounter(limit1: Float, limit2: Float) {
    val min: Float = Math.min(limit1, limit2)
    val max: Float = Math.max(limit1, limit2)
    private var count = 0f

    fun add(x: Float) {
        count += x
    }

    fun reset() {
        count = 0f
    }

    val isTooSmall: Boolean
        get() = count < min

    fun hasDistance(): Boolean {
        return count >= min
    }

    val isTooLarge: Boolean
        get() = count > max
}
